### PR TITLE
Added initial implementation of registry component and refactored events and sessions to work with the new regstry

### PR DESCRIPTION
**Pull request by pzaino**
This PR contains:

Go Plugins interface
Amass Plugins Registry
Events and Sessions refactoring to work with the new registry

### CODE REVIEW

File: registry/registry.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+	} else {
+		if _, ok := r.HandlersMap[h.EventType][h.Name]; ok {
+			return fmt.Errorf("Handler %s already registered for EventType %d", h.Name, h.EventType)
+		}
+	}
+	return nil
+}
+
+// Returns a list of handlers for a given event type
+func (r *Registry) GetHandlers(eventType events.EventType) ([]func(*events.Event) error, error) {
```
Comment (caffix) [inline registry/registry.go:107]: The GetHandlers method will need to return a slice of Handler structs

File: registry/datattypes.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+}
+
+// Each plugins must return an Handlers list at initPlugin() time
+// so that we can determine which events should be sent to the plugin
+// and which handlers should be called for each event.
+
+type Handler struct {
+	Name      string
+	EventType events.EventType
+	Transform []string
```
Comment (caffix) [inline registry/datattypes.go:19]: We should change Transform to Transforms due to the field being a slice

Comment (pzaino) [reply registry/datattypes.go:19]: Fixed :)

File: registry/registry.go (hunk 1 above)
Comment (pzaino) [reply registry/registry.go]: Ok, cracked it :)

On new commit GetHandlers returns a map of slices of Hndler structs where the map key is a single transformation, so  it's possible to retrieve a list of Handler structs using a transformation as key. Hope this is "best of both worlds".

**Review by caffix — APPROVED** [review]
Approved

**Pull request merged**